.ci/test_wrapper.sh

```diff
@@ -1,38 +1,38 @@
 #!/bin/bash
 set -x
 
-KEYLIME_HOME="${KEYLIME_HOME=:-/root/keylime}"
-
 # Configure swtpm2
-cd ${KEYLIME_HOME}/scripts
-chmod +x init_tpm_server
-chmod +x tpm_serverd
-install -c tpm_serverd /usr/local/bin/tpm_serverd
-install -c init_tpm_server /usr/local/bin/init_tpm_server
-# Server needs to be running, or tpm2-abrmd.service will fail.
-/usr/local/bin/tpm_serverd
+mkdir /tmp/tpmdir
+swtpm_setup --tpm2 \
+     --tpmstate /tmp/tpmdir \
+     --createek --decryption --create-ek-cert \
+     --create-platform-cert \
+     --display
+swtpm socket --tpm2 \
+     --tpmstate dir=/tmp/tpmdir \
+     --flags startup-clear \
+     --ctrl type=tcp,port=2322 \
+     --server type=tcp,port=2321 \
+     --daemon
+export TPM2TOOLS_TCTI=tabrmd:
 
-export TPM2TOOLS_TCTI="tabrmd:bus_name=com.intel.tss2.Tabrmd"
-pkill -HUP dbus-daemon
+# Configure dbus
+sudo rm -rf /var/run/dbus
+sudo mkdir /var/run/dbus
+sudo dbus-daemon --system
 
-# Configure tpm2-abrmd systemd
-# systemd-udev-settle is not needed to run tpm2-tools in container
-sed -i 's/^After/#&/' /usr/lib/systemd/system/tpm2-abrmd.service
-sed -i 's/^Requires/#&/' /usr/lib/systemd/system/tpm2-abrmd.service
-# the tpm device is not needed to run tpm2-tools in container with the
-# tpm emulator
-sed -i 's/^ConditionPathExists/#&/' /usr/lib/systemd/system/tpm2-abrmd.service
-sed -i 's/.*ExecStart.*/ExecStart=\/usr\/sbin\/tpm2-abrmd --tcti=mssim/' /usr/lib/systemd/system/tpm2-abrmd.service
-systemctl daemon-reload
-systemctl enable tpm2-abrmd
-systemctl start tpm2-abrmd
-# Check that tpm2-abrmd is actually running as starting it won't report failures
-if ! (systemctl -q is-active tpm2-abrmd); then
-    echo "tpm2-abrmd failed to start, exiting"
-    exit 1
-fi
+tpm2-abrmd \
```

Comment:
@lkatalin or @maugustosilva Do you know if the tpm2-abrmd stuff is needed? I thought newer kernels have their own internal resource manager for this kind of stuff.

Reply:
@mpeters I don't believe it's necessary unless you are using swtpm with the `socket` option. I've been using it with `swtpm chardev` instead and with the kernel resource manager `/dev/tpmrm0`:

```
modprobe tpm_vtpm_proxy
swtpm_setup --tpm2 \
          --tpmstate /var/vtpm \
          --createek --decryption --create-ek-cert \
          --create-platform-cert \
          --display \
swtpm chardev --tpm2 --vtpm-proxy --tpmstate dir=/var/vtpm -d
export TPM2TOOLS_TCTI="device:/dev/tpmrm0"
export TCTI="device:/dev/tpmrm0"
```

If you do it this way, you also don't need the `dbus-daemon`.

Note: this assumes it is running in a VM without a hardware tpm attached (otherwise I think the device defaults to `/dev/tpmrm1` with the swtpm since the hardware tpm is always tpm0, and this doesn't work well for reasons that I forget).

Reply:
As for whether swtpm socket or swtpm chardev will work better for testing, that I am not sure.

Reply:
Ok, we'll just go with socket and abrmd for now